### ISSUE TITLE
[epoxy-paging] Add way of clearing the model cache

### ISSUE THREAD

**Issue by kakai248** [airbnb/epoxy#583]
I have a paged controller with data that comes from my database. The screen where this controller is used allows selecting multiple items. In the ViewModel I keep a set of items that are selected. To be able to diff properly when an item is selected/deselected, I keep a boolean in the object that I pass to the epoxy model. Each time an item is selected/deselected, I do `requestModelBuild()`. Since all the items are cached, there are no changes.

Model cache only reacts to updates in the PagedList, which I have no control of. It would be good if we were able to manually clear the cache before requesting another model build.

**Comment by elihart** [comment]
This seems reasonable - I'd be open to a pull request if you want to attempt it yourself (or anyone else)

**Comment by kakai248** [comment]
I'll try to open the PR during the weekend. Unless someone wants to do it first.

I was thinking of adding a method in the model cache that nulls all the values and a method in the controller that calls the cache method and then requests a model build (something like `requestForcedModelBuild()`, in lack of a better name).

What do you think?

**Comment by elihart** [comment]
that sounds reasonable to me :)

yigit would know best, since he originally wrote it, but he probably doesn't have much time to help maintain and expand this.

**Comment by baoti** [comment]
It looks like #634. You can custom `itemDiffCallback`:

```
override fun areContentsTheSame(oldItem: Any, newItem: Any) = oldItem == newItem && isSelected(oldItem) == isSelected(newItem)
```

then, pass the `itemDiffCallback` to PagedListEpoxyController.

**Comment by kakai248** [comment]
@baoti Won't work. Diffs are only done after the models get rebuilt. And the models related to the PagedList elements are only rebuilt if the DataSource invalidates, otherwise they are cached. My issue is that I want to rebuild the models without the DataSource elements changing.

**Comment by elihart** [comment]
Merged - thanks for the help!

**Comment by NapasPayu** [comment]
@elihart May I know when this is going to release? Thanks

**Comment by elihart** [comment]
@NapasPayu I'll try to release it in the next few days

**Comment by NapasPayu** [comment]
@elihart I am still waiting for this. Please help push it ^^

**Comment by kakai248** [comment]
It was released with 3.2.0
